dbms/src/Parsers/ExpressionListParsers.cpp

```diff
@@ -623,4 +623,71 @@ bool ParserIntervalOperatorExpression::parseImpl(Pos & pos, ASTPtr & node, Expec
 }
 
 
+bool ParserKeyValuePair::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserIdentifier id_parser;
+    ParserLiteral literal_parser;
+    ParserExpression expression_parser;
+
+    ASTPtr identifier;
+    ASTPtr value;
+    if (!id_parser.parse(pos, identifier, expected))
+        return false;
+
+    if (!expression_parser.parse(pos, value, expected))
+        return false;
+
+    auto pair = std::make_shared<ASTPair>();
+    pair->first = Poco::toLower(typeid_cast<ASTIdentifier &>(*identifier.get()).name);
+    pair->second = value;
+    node = pair;
+    return true;
+}
+
+
+class ParserKeyValueFunctionPair : public IParserBase
+{
+protected:
+    const char * getName() const override { return "stacked parser"; }
```

Comment:
It's because the logic of this parser is the next:
1) Try parse key-value function
2) If 1 failed then try to parse key-value pair
This parser combine two parsers. Maybe "stacked parser" wasn't good name